src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2346,7 +2347,26 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False):  #
         _create_new_asp = True
     else:
         logger.warning("Resource group '%s' already exists.", rg_name)
-        _create_new_asp = should_create_new_asp(cmd, rg_name, asp, location)
+        _asp_generic = asp[:-len(asp.split("_")[4])]
```

Comment:
would `'_'.join(asp.split('_', 4)[:4])`  better?

Reply:
is that more performant - I was using c# LINQ equivalent Python methods. I thought this was easy to understand logic wise but I am no Python expert :)

Reply:
my suggestion is based on being more readable